tiledb/sm/kv/kv.cc

```diff
@@ -541,61 +547,30 @@ Status KV::prepare_read_attributes() {
   read_attribute_var_sizes_.resize(read_attribute_num_);
   read_attribute_types_.resize(read_attribute_num_);
 
-  unsigned int i = 0, aid;
+  unsigned int i = 0;
   for (const auto& attr : attributes_) {
     read_attributes_[i] = new (std::nothrow) char[attr.size() + 1];
     strcpy(read_attributes_[i], attr.data());
-    RETURN_NOT_OK(schema_->attribute_id(attr, &aid));
-    read_attribute_types_[i] = schema_->attribute(aid)->type();
-    read_attribute_var_sizes_[i++] = schema_->var_size(aid);
+    read_attribute_types_[i] = schema_->type(attr);
+    read_attribute_var_sizes_[i++] = schema_->var_size(attr);
   }
 
   return Status::Ok();
 }
 
 Status KV::prepare_write_attributes() {
   // +3 because of the two special key attributes and coordinates
```

Comment:
Looks like this comment line should be removed

Reply:
Good catch, thanks!